src/controllers/authController.js

```diff
@@ -0,0 +1,41 @@
+import bcrypt from "bcrypt"
+
+import db from "./../db/db.js"
+import { signupSchema } from "./../schemas/authSchema.js"
+
+export async function signup(req, res) {
+  const { error } = signupSchema.validate(req.body, { abortEarly: false })
+
+  if (error) {
+    const messages = error.details.map(({ message }) => message).join(", ")
+    return res.status(422).send(messages)
+  }
+
+  try {
+    const { name, email, password } = req.body
+
+    const emailExists = await db.collection("users").findOne({ email })
+
+    if (emailExists) return res.status(422).send("This email already exists.")
```

Comment:
Quando for retornar os erros, tenta deixar por padrão a primeira palavra já atrelada onde deu o erro pra ficar mais facil de identificar no front
exemplo:
"email", already exists
deixando nesse padrão a gente consegue, por exemplo mostrar mais facil pro usuario oq ta errado no front

Reply:
Boa sugestão!